shell/platform/linux/fl_scrolling_manager.cc

```diff
@@ -0,0 +1,175 @@
+      g_object_new(fl_scrolling_manager_get_type(), nullptr));
+
+  self->view_delegate = view_delegate;
+  self->pan_started = false;
+  self->zoom_started = false;
+  self->rotate_started = false;
+
+  return self;
+}
+
+void fl_scrolling_manager_set_last_mouse_position(FlScrollingManager* manager,
+                                                  gdouble x,
+                                                  gdouble y) {
+  manager->last_x = x;
+  manager->last_y = y;
+}
+
+void fl_scrolling_manager_handle_scroll_event(FlScrollingManager* manager,
+                                              GdkEventScroll* event,
+                                              gint scale_factor) {
+  gdouble scroll_delta_x = 0.0, scroll_delta_y = 0.0;
+  if (event->direction == GDK_SCROLL_SMOOTH) {
+    scroll_delta_x = event->delta_x;
+    scroll_delta_y = event->delta_y;
+  } else if (event->direction == GDK_SCROLL_UP) {
+    scroll_delta_y = -1;
+  } else if (event->direction == GDK_SCROLL_DOWN) {
+    scroll_delta_y = 1;
+  } else if (event->direction == GDK_SCROLL_LEFT) {
+    scroll_delta_x = -1;
+  } else if (event->direction == GDK_SCROLL_RIGHT) {
+    scroll_delta_x = 1;
+  }
+
+  // The multiplier is taken from the Chromium source
+  // (ui/events/x/events_x_utils.cc).
+  const int kScrollOffsetMultiplier = 53;
+  scroll_delta_x *= kScrollOffsetMultiplier * scale_factor;
+  scroll_delta_y *= kScrollOffsetMultiplier * scale_factor;
+
+  if (gdk_device_get_source(gdk_event_get_source_device((GdkEvent*)event)) ==
+      GDK_SOURCE_TOUCHPAD) {
+    scroll_delta_x *= -1;
+    scroll_delta_y *= -1;
+    if (event->is_stop) {
+      fl_scrolling_view_delegate_send_pointer_pan_zoom_event(
+          manager->view_delegate, event->time * kMicrosecondsPerMillisecond,
+          event->x * scale_factor, event->y * scale_factor, kPanZoomEnd,
+          manager->pan_x, manager->pan_y, 0, 0);
+      manager->pan_started = FALSE;
+    } else {
+      if (!manager->pan_started) {
+        manager->pan_x = 0;
+        manager->pan_y = 0;
+        fl_scrolling_view_delegate_send_pointer_pan_zoom_event(
+            manager->view_delegate, event->time * kMicrosecondsPerMillisecond,
+            event->x * scale_factor, event->y * scale_factor, kPanZoomStart, 0,
+            0, 0, 0);
+        manager->pan_started = TRUE;
+      }
+      manager->pan_x += scroll_delta_x;
+      manager->pan_y += scroll_delta_y;
+      fl_scrolling_view_delegate_send_pointer_pan_zoom_event(
+          manager->view_delegate, event->time * kMicrosecondsPerMillisecond,
+          event->x * scale_factor, event->y * scale_factor, kPanZoomUpdate,
+          manager->pan_x, manager->pan_y, 1, 0);
+    }
+  } else {
+    manager->last_x = event->x * scale_factor;
+    manager->last_y = event->y * scale_factor;
+    fl_scrolling_view_delegate_send_mouse_pointer_event(
+        manager->view_delegate,
+        FlutterPointerPhase::kMove /* will be overridden */,
```

Comment:
It's not clear to me from this comment why this will be overridden and why that matters - perhaps can be elaborated a bit?

Reply:
I expanded the comment. Let me know if it makes sense now.

Reply:
Yes, thanks!